docs/developer-guide/join-streams-and-tables.rst

```diff
@@ -0,0 +1,329 @@
+.. _join-streams-and-tables:
+
+Join Streams and Tables with KSQL
+=================================
+
+You can use KSQL to merge streams of data in realtime by using a SQL-like
+*join* syntax. A join in KSQL is similar to a join clause in a relational
+database, which combines data from two sources based on values common to
+both. The result of a KSQL join is a new stream or table that's populated
+with the column values that you specify in a SELECT statement.
+
+KSQL automates creating Kafka Streams applications. You don’t need to write
+the low-level logic around joining streams, so you can focus on the business
+logic for combining your streaming data.
+
+You can join streams and tables in three different ways:
+
+* Join two streams to create a new stream.
+* Join two tables to create a new table.
+* Join a stream and a table to create a new stream.
+
+JOIN Clause
+***********
+
+The KSQL JOIN clause has the familiar syntax of a SQL JOIN clause.
+The following example creates a ``pageviews_enriched`` stream, which is a
+combination of a ``pageviews`` stream and a ``users`` table:
+
+.. code:: sql
+
+   CREATE STREAM pageviews_enriched AS
+     SELECT users.userid AS userid, pageid, regionid, gender FROM pageviews
+     LEFT JOIN users ON pageviews.userid = users.userid;
+
+
+For the full code example, see :ref:`ksql_quickstart-docker`.
+
+Here's an example stream-stream join that combines a ``shipments`` stream with

```

Comment:
I know that this is lifted from some demos, but I don't think this is a great example because of the current join semantics.

Essentially, if the orders event arrives first (which it mostly will do), there will always be one tuple in `late_orders` with `(s.orderid == NULL, o.orderid != NULL`). If that is the _only_ tuple, then you have a late order. But if the shipment was made within 2 hours, then you will get a tuple with `(s.orderid != NULL, o.orderid != NULL)`.

I will leave it to @rmoff and @miguno to suggest a better example of a stream-stream join.